addons/purchase_stock/tests/test_purchase_order_suggest.py

```diff
@@ -441,3 +451,29 @@ def test_purchase_order_suggest_quantities_multiwarehouse(self):
         self.assertRecordValues(po_2.order_line, [
             {'product_id': self.product_1.id, 'product_qty': 10},
         ])
+
+    def test_purchase_order_suggest_access_error_non_admin(self):
+        """ Test that non-admin users can use the suggest feature without access errors """
+        today = fields.Datetime.now()
+        warehouse = self.delivery_type.warehouse_id
+        self.env['stock.quant']._update_available_quantity(self.product_1, warehouse.lot_stock_id, 50)
+        self._create_and_process_delivery_at_date(
+            [(self.product_1, 10)], today - relativedelta(months=1), warehouse=warehouse
+        )
+        self.env['stock.quant']._update_available_quantity(self.product_1, warehouse.lot_stock_id, -45)
```

Comment:
I don't understand why you add stock then update the stock in negative ?
If you want to create a demand, just add less quantity in stock
```suggestion
        self.env['stock.quant']._update_available_quantity(self.product_1, warehouse.lot_stock_id, 15)
        self._create_and_process_delivery_at_date(
            [(self.product_1, 10)], today - relativedelta(months=1), warehouse=warehouse
        )
```